db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBClient.java

```diff
@@ -117,88 +176,173 @@ public CompletableFuture<StreamMetadata> getStreamMetadata(String streamName, Re
         });
     }
 
-    public CompletableFuture<ReadResult> readAll() {
-        return this.readAll(Long.MAX_VALUE, ReadAllOptions.get());
-    }
-
-    public CompletableFuture<ReadResult> readAll(long maxCount) {
-        return this.readAll(maxCount, ReadAllOptions.get());
-    }
-
-    public CompletableFuture<ReadResult> readAll(ReadAllOptions options) {
-        return this.readAll(Long.MAX_VALUE, options);
+    /**
+     * Reads events from the $all stream. The reading can be done forward and backward.
+     */
+    public CompletableFuture<List<ResolvedEvent>> readAll() {
+        return this.readAll(ReadAllOptions.get());
     }
 
-    public CompletableFuture<ReadResult> readAll(long maxCount, ReadAllOptions options) {
-        return readEventsFromPublisher(this.readAllReactive(maxCount, options));
+    /**
+     * Reads events from the $all stream. The reading can be done forward and backward.
+     * @param options options of the read $all request.
+     */
+    public CompletableFuture<List<ResolvedEvent>> readAll(ReadAllOptions options) {
+        return readEventsFromPublisher(this.readAllReactive(options));
     }
 
+    /**
+     * Reads events from the $all stream. The reading can be done forward and backward.
+     */
     public Publisher<ResolvedEvent> readAllReactive() {
-        return this.readAllReactive(Long.MAX_VALUE, ReadAllOptions.get());
-    }
-
-    public Publisher<ResolvedEvent> readAllReactive(long maxCount) {
-        return this.readAllReactive(maxCount, ReadAllOptions.get());
+        return this.readAllReactive(ReadAllOptions.get());
     }
 
+    /**
+     * Reads events from the $all stream. The reading can be done forward and backward.
+     * @param options options of the read $all request.
+     */
     public Publisher<ResolvedEvent> readAllReactive(ReadAllOptions options) {
-        return this.readAllReactive(Long.MAX_VALUE, options);
-    }
-
-    public Publisher<ResolvedEvent> readAllReactive(long maxCount, ReadAllOptions options) {
         if (options == null)
             options = ReadAllOptions.get();
 
-        if (!options.hasUserCredentials())
-            options.authenticated(this.credentials);
-
-        return new ReadAll(this.client, maxCount, options);
-    }
-
+        return new ReadAll(this.getGrpcClient(), options);
+    }
+
+    /**
+     * Subscribes to a given stream. This kind of subscription specifies a starting point (by default, the beginning of
```

Comment:
**SUGGESTION:** Maybe you could use text from gRPC clients docs?

> Subscriptions allow you to subscribe to a stream and receive notifications about new events added to the stream.
> 
> You provide an event handler and an optional starting point to the subscription. The handler is called for each event from the starting point onward.
> 
> If events already exist, the handler will be called for each event one by one until it reaches the end of the stream. From there, the server will notify the handler whenever a new event appears.

https://developers.eventstore.com/clients/grpc/subscriptions.html#catch-up-subscriptions

Reply:
Done